src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -510,14 +505,18 @@ private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase)
      * possible match is within a comment.
      *
      * @param pattern The regular expression pattern to use.
-     * @param line The line of test to do the match on.
      * @param lineNo The line number in the file.
      * @return True if a match was found inside a comment.
      */
     // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
     @SuppressWarnings("deprecation")
-    private boolean matchesComment(Pattern pattern, String line, int lineNo) {
-        final Matcher matcher = pattern.matcher(line);
+    private boolean matchesComment(Pattern pattern, int lineNo) {
+        // Remember: The lines number returned from the AST is 1-based, but
+        // the lines number in this array are 0-based. So you will often
+        // see a "lineNo-1" etc.
+        final String[] lines = getLines();
```

Comment:
As silly as this sounds, it might be best to pass code points to this method, and convert to a string to avoid using `getLines`.

If nothing else, we could use `getLine`, no need to get all lines in file.

@strkkk what do you think?

Reply:
Yes, that sounds reasonable. It is better to avoid unnecessary conversions

Reply:
done

Reply:
Resolved.